src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -112,141 +168,254 @@ abstract class LazyModule()(implicit val p: Parameters)
+    // Generate [[AutoBundle]] IO from [[forward]].
     val auto = IO(new AutoBundle(forward.map { d => (d.name, d.data, d.flipped) }:_*))
+    // Pass the [[Dangle]]s which remained and were used to generate the [[AutoBundle]] I/O ports up to the [[parent]] [[LazyModule]]
     val dangles = (forward zip auto.elements) map { case (d, (_, io)) =>
       if (d.flipped) { d.data <> io } else { io <> d.data }
       d.copy(data = io, name = wrapper.suggestedName + "_" + d.name)
     }
+    // Push all [[LazyModule.inModuleBody]] to [[chisel3.internal.Builder]].
     wrapper.inModuleBody.reverse.foreach { _() }
+    // Return [[IO]] and [[Dangle]] of this [[LazyModuleImp]].
     (auto, dangles)
   }
 
+  /** Ask each [[BaseNode]] in [[wrapper.nodes]] to call [[BaseNode.finishInstantiate]]
+    * Note: There are 2 different `finishInstantiate` methods:
+    *   [[LazyModuleImp.finishInstantiate]] and [[BaseNode.finishInstantiate]],
+    *   the former is a wrapper to the latter.
+    * */
   protected[diplomacy] def finishInstantiate() {
     wrapper.nodes.reverse.foreach { _.finishInstantiate() }
   }
 }
 
+/** 
+ * 
+ */
 class LazyModuleImp(val wrapper: LazyModule) extends MultiIOModule with LazyModuleImpLike {
+  /** Instantiate hardware of this `Module`. */
   val (auto, dangles) = instantiate()
 }
 
+/** 
+ * 
+ */
 class LazyRawModuleImp(val wrapper: LazyModule) extends RawModule with LazyModuleImpLike {
-  // These wires are the default clock+reset for all LazyModule children
-  // It is recommended to drive these even if you manually shove most of your children
-  // Otherwise, anonymous children (Monitors for example) will not be clocked
+  // These wires are the default clock+reset for all LazyModule children.
+  // It is recommended to drive these even if you manually drive the [[clock]] and [[reset]] of all of the [[LazyRawModuleImp]] children.
+  // Otherwise, anonymous children ([[Monitor]]s for example) will not have their [[clock]] and/or [[reset]] driven properly.
+  /** drive clock explicitly. */
   val childClock = Wire(Clock())
+  /** drive reset explicitly. */
   val childReset = Wire(Bool())
+  // the default is that these are disabled
   childClock := Bool(false).asClock
   childReset := Bool(true)
   val (auto, dangles) = withClockAndReset(childClock, childReset) {
     instantiate()
   }
 }
 
+/** Used for a [[LazyModule]] which does not need to define any [[LazyModuleImp]] implementation.
+  * It can be used as wrapper that only instantiates and connects [[LazyModule]]s.
+  * */
 class SimpleLazyModule(implicit p: Parameters) extends LazyModule
 {
   lazy val module = new LazyModuleImp(this)
 }
 
+/** 
+ * 
+ */
 trait LazyScope
 {
   this: LazyModule =>
   override def toString: String = s"LazyScope named $name"
+  /** Preserve the previous value of the [[LazyModule.scope]], because when calling [[apply]] function,
+    * [[LazyModule.scope]] will be altered.
+    * */
   def apply[T](body: => T) = {
```

Comment:
```suggestion
/** Allows dynamic creation of [[Module]] hierarchy and "shoving" logic into a [[LazyModule]]. */ 
trait LazyScope
{
  this: LazyModule =>
  override def toString: String = s"LazyScope named $name"
  
  /** Evaluate `body` in the current [[LazyModule.scope]] */
  def apply[T](body: => T) = {
    // Preserve the previous value of the [[LazyModule.scope]], because when calling [[apply]] function,
    // [[LazyModule.scope]] will be altered.
```